src/scene-entry-manager.js

```diff
@@ -397,52 +397,30 @@ export default class SceneEntryManager {
     let currentVideoShareEntity;
     let isHandlingVideoShare = false;
 
-    const shareVideoMediaStream = async (constraints, isDisplayMedia, target) => {
-      if (isHandlingVideoShare) return;
-      isHandlingVideoShare = true;
-
-      let newStream;
-
-      try {
-        if (isDisplayMedia) {
-          newStream = await navigator.mediaDevices.getDisplayMedia(constraints);
-        } else {
-          newStream = await navigator.mediaDevices.getUserMedia(constraints);
-        }
-      } catch (e) {
-        isHandlingVideoShare = false;
-        this.scene.emit("share_video_failed");
-        return;
-      }
-
-      const videoTracks = newStream ? newStream.getVideoTracks() : [];
-
-      if (videoTracks.length > 0) {
-        newStream.getVideoTracks().forEach(track => mediaStream.addTrack(track));
-
-        if (newStream && newStream.getAudioTracks().length > 0) {
-          const audioSystem = this.scene.systems["hubs-systems"].audioSystem;
-          audioSystem.addStreamToOutboundAudio("screenshare", newStream);
-        }
+    const shareSuccess = (isDisplayMedia, isVideoTrackAdded) => {
+      isHandlingVideoShare = false;
 
-        await NAF.connection.adapter.setLocalMediaStream(mediaStream);
+      if (isVideoTrackAdded) {
+        currentVideoShareEntity = spawnMediaInfrontOfPlayer(this.mediaDevicesManager.mediaStream, undefined);
 
-        if (target === "avatar") {
-          this.avatarRig.setAttribute("player-info", { isSharingAvatarCamera: true });
-        } else {
-          currentVideoShareEntity = spawnMediaInfrontOfPlayer(mediaStream, undefined);
+        // Wire up custom removal event which will stop the stream.
+        currentVideoShareEntity.setAttribute("emit-scene-event-on-remove", "event:action_end_video_sharing");
 
-          // Wire up custom removal event which will stop the stream.
-          currentVideoShareEntity.setAttribute("emit-scene-event-on-remove", "event:action_end_video_sharing");
-        }
+        this.scene.emit("share_video_enabled", { source: isDisplayMedia ? "screen" : "camera" });
+        this.scene.addState("sharing_video");
       }
+    };
 
-      this.scene.emit("share_video_enabled", { source: isDisplayMedia ? "screen" : "camera" });
-      this.scene.addState("sharing_video");
+    const shareError = error => {
+      console.error(error);
       isHandlingVideoShare = false;
+      this.scene.emit("share_video_failed");
     };
 
-    this.scene.addEventListener("action_share_camera", event => {
+    this.scene.addEventListener("action_share_camera", () => {
+      if (isHandlingVideoShare) return;
+      isHandlingVideoShare = true;
+
       const constraints = {
         video: {
           width: isIOS ? { max: 1280 } : { max: 1280, ideal: 720 },
```

Comment:
We may want to revisit these. A bit afraid of encoding multiple high resolution stream on a phone for example.

Reply:
I think the next step towards better video sharing in general is to expose options to do HD, SD,... so the user can choose that or even switch between them real time. We can adjust the different setups when implementing that. I'll open an issue for that.

Reply:
https://github.com/mozilla/hubs/issues/3989